js/src/tests/llm_evaluator.int.test.ts

```diff
@@ -0,0 +1,183 @@
+});
+
+test("llm evaluator with custom variable mapping", async () => {
+  const evaluator = await LLMEvaluator.create({
+    promptTemplate: [
+      [
+        "system",
+        "Is the output accurate with respect to the context and question? Y/N",
+      ],
+      ["human", "Context: {context}\nQuestion: {question}\nOutput: {output}"],
+    ],
+    scoreConfig: new CategoricalScoreConfig({
+      key: "accuracy",
+      choices: ["Y", "N"],
+      description:
+        "Whether the output is accurate with respect to the context and question.",
+      reasoningKey: "explanation",
+      reasoningDescription: "First, think step by step to explain your score.",
+    }),
+    chatModel: CHAT_MODEL,
+    mapVariables: (run: any, example?: any) => ({
+      context: example?.inputs?.context || "",
+      question: example?.inputs?.question || "",
+      output: run.outputs?.output || "",
+    }),
+  });
+
+  expect(evaluator).toBeDefined();
+});
+
+test("llm evaluator can evaluate runs", async () => {
+  const client = new Client();
+  await client.clonePublicDataset(
+    "https://beta.smith.langchain.com/public/06785303-0f70-4466-b637-f23d38c0f28e/d",
+    {
+      datasetName: TESTING_DATASET_NAME,
+    }
+  );
+  const evaluator = await LLMEvaluator.create({
+    promptTemplate: "Is the response vague? Y/N\n{response}",
+    scoreConfig: new CategoricalScoreConfig({
+      key: "vagueness",
+      choices: ["Y", "N"],
+      description: "Whether the response is vague. Y for yes, N for no.",
+      reasoningKey: "explanation",
+      reasoningDescription: "First, think step by step to explain your score.",
+    }),
+    chatModel: CHAT_MODEL,
+    mapVariables: (run: any, _example?: any) => ({
+      response: run.outputs?.["output"] ?? "",
+    }),
+  });
+
+  const targetFunc = (input: Record<string, any>) => {
+    return { output: input.question + " This is a test response" };
+  };
+
+  const evalRes = await evaluate(targetFunc, {
+    data: TESTING_DATASET_NAME,
+    evaluators: [evaluator],
+    description: "LLM evaluator test run",
+  });
+
+  expect(evalRes.results).toHaveLength(10);
+  const firstResult = evalRes.results[0];
+
+  const evaluation = firstResult.evaluationResults.results[0];
+  expect(evaluation.key).toBe("vagueness");
+  expect(["Y", "N"]).toContain(evaluation.value);
+  expect(evaluation.comment).toBeDefined();
+
+  await client.deleteDataset({ datasetName: TESTING_DATASET_NAME });
+});
+
+test("llm evaluator with multiple prompt messages", async () => {
+  const evaluator = await LLMEvaluator.create({
+    promptTemplate: [
+      ["system", "You are a helpful assistant evaluating responses."],
+      ["human", "Rate this response from 0 to 1: {response}"],
+    ],
+    scoreConfig: new ContinuousScoreConfig({
+      key: "rating",
+      description: "Quality rating from 0 to 1",
+      min: 0,
+      max: 1,
+    }),
+    chatModel: CHAT_MODEL,
+    mapVariables: (run: any, _example?: any) => ({
```

Comment:
Is there a better name for this we can use? `mapVariables` sounds like a map of variables or something and `.map()` has a specific meaning in JS (and most languages?)

Reply:
Open to suggestions, this is what we use in python, agree it is somewhat confusing. The true purposes is "map variables from your example/run to variables used in your evaluator prompt" - not sure how to best condense this into a succinct variable name.

Reply:
`formatPrompt`?

Reply:
Obviously would need an update on Python side but I think this name is not good